cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis.hs

```diff
@@ -0,0 +1,1310 @@
+    liftIO $ writeOutput mOutFile (renderTxIn txin)
+
+
+runGenesisAddrCmd ::
+     VerificationKeyFile In
+  -> NetworkId
+  -> Maybe (File () Out)
+  -> ExceptT ShelleyGenesisCmdError IO ()
+runGenesisAddrCmd vkeyPath network mOutFile = do
+    vkey <- firstExceptT ShelleyGenesisCmdTextEnvReadFileError . newExceptT $
+            readFileTextEnvelope (AsVerificationKey AsGenesisUTxOKey) vkeyPath
+    let vkh  = verificationKeyHash (castVerificationKey vkey)
+        addr = makeShelleyAddress network (PaymentCredentialByKey vkh)
+                                  NoStakeAddress
+    liftIO $ writeOutput mOutFile (serialiseAddress addr)
+
+writeOutput :: Maybe (File () Out) -> Text -> IO ()
+writeOutput (Just (File fpath)) = Text.writeFile fpath
+writeOutput Nothing                   = Text.putStrLn
+
+
+--
+-- Create Genesis command implementation
+--
+
+runGenesisCreateCmd
+  :: KeyOutputFormat
+  -> GenesisDir
+  -> Word  -- ^ num genesis & delegate keys to make
+  -> Word  -- ^ num utxo keys to make
+  -> Maybe SystemStart
+  -> Maybe Lovelace
+  -> NetworkId
+  -> ExceptT ShelleyGenesisCmdError IO ()
+runGenesisCreateCmd
+    fmt (GenesisDir rootdir)
+    genNumGenesisKeys genNumUTxOKeys
+    mStart mAmount network = do
+
+  liftIO $ do
+    createDirectoryIfMissing False rootdir
+    createDirectoryIfMissing False gendir
+    createDirectoryIfMissing False deldir
+    createDirectoryIfMissing False utxodir
+
+  template <- readShelleyGenesisWithDefault (rootdir </> "genesis.spec.json") adjustTemplate
+  alonzoGenesis <- readAlonzoGenesis (rootdir </> "genesis.alonzo.spec.json")
+  conwayGenesis <- readConwayGenesis (rootdir </> "genesis.conway.spec.json")
+
+  forM_ [ 1 .. genNumGenesisKeys ] $ \index -> do
+    createGenesisKeys  gendir  index
+    createDelegateKeys fmt deldir index
+
+  forM_ [ 1 .. genNumUTxOKeys ] $ \index ->
+    createUtxoKeys utxodir index
+
+  genDlgs <- readGenDelegsMap gendir deldir
+  utxoAddrs <- readInitialFundAddresses utxodir network
+  start <- maybe (SystemStart <$> getCurrentTimePlus30) pure mStart
+
+  let shelleyGenesis =
+        updateTemplate
+          -- Shelley genesis parameters
+          start genDlgs mAmount utxoAddrs mempty (Lovelace 0) [] [] template
+
+  void $ writeFileGenesis (rootdir </> "genesis.json")        $ WritePretty shelleyGenesis
+  void $ writeFileGenesis (rootdir </> "genesis.alonzo.json") $ WritePretty alonzoGenesis
+  void $ writeFileGenesis (rootdir </> "genesis.conway.json") $ WritePretty conwayGenesis
+  --TODO: rationalise the naming convention on these genesis json files.
+  where
+    adjustTemplate t = t { sgNetworkMagic = unNetworkMagic (toNetworkMagic network) }
+    gendir  = rootdir </> "genesis-keys"
+    deldir  = rootdir </> "delegate-keys"
+    utxodir = rootdir </> "utxo-keys"
+
+toSKeyJSON :: Key a => SigningKey a -> ByteString
+toSKeyJSON = LBS.toStrict . textEnvelopeToJSON Nothing
+
+toVkeyJSON ::
+#if __GLASGOW_HASKELL__ >= 904
```

Comment:
this shouldn't be needed anymore after https://github.com/input-output-hk/cardano-cli/pull/245